### PR TITLE
Remove api.AudioNode.disconnect.*_parameter from BCD

### PR DESCRIPTION
**Pull request by queengooborg**
This PR removes the `disconnect.*_parameter` member of the `AudioNode` API from BCD. Per the [data guidelines](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#removal-of-irrelevant-features), this feature can be considered irrelevant and may be removed from BCD accordingly. Even if the current data suggests that the feature is supported, lack of support has been confirmed by the [mdn-bcd-collector](https://mdn-bcd-collector.gooborg.com) project (v9.0.2).

Tests Used:
https://mdn-bcd-collector.gooborg.com/tests/api/AudioNode/disconnect/destination_parameter
https://mdn-bcd-collector.gooborg.com/tests/api/AudioNode/disconnect/input_parameter

Additional Notes: The test function tests to see if the argument has been accessed.  While this may produce false positives (especially in Chrome) when the value is checked but doesn't influence behavior, I can't think of a situation where a false negative could occur.

### CODE REVIEW

**Comment by queengooborg** [comment]
Hey @tomayac, maybe you have some insight into this one as well, if you'd be down to review these changes?

**Comment by tomayac** [comment]
I'd defer to @hoch as the domain expert.

**Comment by hoch** [comment]
> Even if the current data suggests that the feature is supported, lack of support has been confirmed by the mdn-bcd-collector project (v9.0.2).

Could you clarify what this means? We have an [IDL](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/webaudio/audio_node.idl) and [associated WPTs](https://wpt.fyi/results/webaudio/the-audio-api/the-audionode-interface?label=experimental&label=master&aligned) to test this feature. I am not familiar with BCD and really can't tell the meaning of its confirmation.

**Comment by queengooborg** [comment]
Thanks for asking, @hoch!

The TLDR is that the `mdn-bcd-collector` project (available at https://mdn-bcd-collector.gooborg.com/) is a website and service that runs predefined test code to check if a feature is supported or not.  In this way, it functions similar to WPT, but tailored more for BCD's structure and designed to run on much as old of browsers as possible, and tailored for BCD's tracked features (including non-standard features that WPT doesn't track).

The test code that is used to determine whether these properties are supported are linked above, but I realize that hides a lot of the backend that comes from the helper function which actually tests if the variable has been accessed, `bcd.testOptionParam()` -- that code can be found here:

https://github.com/GooborgStudios/mdn-bcd-collector/blob/ed5ffe5a0eb9a30a79e929a9c828435c22b75fa3/static/resources/harness.js#L325-L461

In summary, this function converts the property into an object with a getter function that sets a boolean variable, `accessed`, to `true` if the getter is ever called (and then returns the value).  `accessed` is then returned by `bcd.testOptionParam()`.

---

What our tests are showing is that in Chrome, neither the `destination` or `input` parameters are actually being accessed when `AudioNode.disconnect(destination, input)` is called.  As far as I can tell, WPT doesn't seem to test if the parameters have any effect?

What I'm wondering is: does a lack of support in Chrome for either parameter seem accurate?

**Comment by hoch** [comment]
Thanks for the explanation, but I need to understand your question first.

Based on the spec (https://webaudio.github.io/web-audio-api/#AudioNode-methods), we do not have `disconnect(destination, input)` but we have other varieties. Is that what you're asking?

**Comment by queengooborg** [comment]
Apologies, I was assuming that `input` was the second argument as BCD doesn't seem to track `output` -- we'll have to fix that!

What we're running into is that it seems as though all the other forms for the method --  including `disconnect(destination)`, `disconnect(destination, output)` and `disconnect(destination, output, input)` -- are ignored entirely, so calling `disconnect(destination)` behaves the exact same way as `disconnect()` (with no arguments).  In other words, our test code is showing that _the parameters to `.disconnect()` are completely ignored by Chrome._  This means one of two things: either our test code is wrong (which I'd be genuinely surprised about) or Chrome indeed doesn't support these arguments.  I'm hoping to get some help to determine which of these two possibilities is reality.

**Comment by hoch** [comment]
No apologies needed! I am also learning new things here. Thanks for this discussion.

> What we're running into is that it seems as though all the other forms for the method -- including disconnect(destination), disconnect(destination, output) and disconnect(destination, output, input) -- are ignored entirely

I assume that you meant these are ignored by BCD. I can see that all of these are properly tested in the [existing WPTs](https://wpt.fyi/results/webaudio/the-audio-api/the-audionode-interface/audionode-disconnect.html?label=master&label=experimental&aligned). Specifically the following test (from [chromium code search](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/external/wpt/webaudio/the-audio-api/the-audionode-interface/audionode-disconnect.html;l=253)):

```js
        // There is no output #2 in the splitter. An exception should be thrown.
        should(function() {
          splitter.disconnect(gain2, 2);
        }, 'splitter.disconnect(gain2, 2)').throw(DOMException, 'IndexSizeError');

        // The splitter and gain1 are not connected. An exception should be
        // thrown.
        should(function() {
          splitter.disconnect(gain1, 0);
        }, 'splitter.disconnect(gain1, 0)').throw(DOMException, 'InvalidAccessError');

        // The splitter output #0 and the gain3 output #0 are not connected. An
        // exception should be thrown.
        should(function() {
          splitter.disconnect(gain3, 0, 0);
        }, 'splitter.disconnect(gain3, 0, 0)').throw(DOMException, 'InvalidAccessError');

        // The output index is out of bound. An exception should be thrown.
        should(function() {
          splitter.disconnect(merger, 3, 0);
        }, 'splitter.disconnect(merger, 3, 0)').throw(DOMException, 'IndexSizeError');
```

> so calling disconnect(destination) behaves the exact same way as disconnect() (with no arguments).

That is actually working as intended. When no arguments are given, `disconnect()` disconnects every connection of the node. Providing `destination` only can be tested when the node has multiple outgoing connections.

> This means one of two things: either our test code is wrong (which I'd be genuinely surprised about) or Chrome indeed doesn't support these arguments.

I hope my explanation above can be a confirmation that the former is true. Let me know if I am mistaken.

**Comment by queengooborg** [comment]
Thank you for the information, this is a really big help!  This goes to show that our parameter test function may also create false negatives, not just false positives.

I'll see what I can do to refine our test code -- marking this PR as a draft as a reminder!

**Comment by queengooborg** [comment]
After attempting to reverse engineer the test code on WPT, I figured out where the issue with the test is.  I strung together the following code:

```js
let context = new OfflineAudioContext(1, 128, 44100);
let source = context.createBufferSource();
let splitter = context.createChannelSplitter(3);
let sum = context.createGain();
source.connect(splitter);
splitter.connect(sum, 0);
splitter.connect(sum, 1);
splitter.connect(sum, 2);
sum.connect(context.destination);
source.start();
```

And then I ran `.disconnect()` three different ways:

```js
splitter.disconnect({
  get: function() {
    console.log('Got!');
    return 1;
  }
});

splitter.disconnect({
  get: function() {
    console.log('Got!');
    return 5;
  }
});

splitter.disconnect(5);
```

The console log in the getter function is never called, and `.disconnect()` does not throw an exception on the second attempt.  The third attempt, which passes in a raw integer, does work and throws an exception.

In other words, our attempts to build an object with a getter is what's causing the issue.  This genuinely surprises me, but I have an idea on how to work around this now.
